### PR TITLE
fix(search): fix load more

### LINKED ISSUE #35558: [Bug]: Search results broken when requesting to load more results
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When I try to click on the button link "Load more results" there are no any other results shown but the content of the search window in reset. This bug appeared after upgrading from 24.x to 25.x.

image

image

### Steps to reproduce

1. Click on the search icon.
2. Enter the text for the search.
3. See the results.
4. Click the link "Load more results".
5. Observe that the search window is empty.

### Expected behavior

The list of files searched additionally should be shown. 

### Installation method

Community Manual installation with Archive

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Updated to a major version (ex. 22.2.3 to 23.0.1)

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "xx.local",
            "192.168.xxx.232",
            "192.168.yyy.100",
            "127.0.1.1"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "25.0.1.1",
        "overwrite.cli.url": "https:\/\/xx.local\/",
        "htaccess.RewriteBase": "\/",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "default_phone_region": "UA",
        "maintenance": false,
        "enable_previews": true,
        "preview_max_x": 350,
        "preview_max_y": 350,
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "app_install_overwrite": [
            "documentserver_community",
            "twofactor_admin",
            "files_trackdownloads",
            "twofactor_webauthn",
            "files_fulltextsearch",
            "fulltextsearch_elasticsearch",
            "fulltextsearch",
            "group_default_quota"
        ],
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpsecure": "ssl",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "theme": "",
        "loglevel": 3,
        "allow_local_remote_servers": true,
        "activity_use_cached_mountpoints": true,
        "trashbin_retention_obligation": "auto",
        "onlyoffice": {
            "verify_peer_off": true,
            "jwt_secret": "secret",
            "jwt_header": "AuthorizationJwt"
        },
        "enabledPreviewProviders": [
            "OC\\Preview\\MP3",
            "OC\\Preview\\TXT",
            "OC\\Preview\\MarkDown",
            "OC\\Preview\\OpenDocument",
            "OC\\Preview\\Krita",
            "OC\\Preview\\Imaginary"
        ],
        "preview_imaginary_url": "http:\/\/localhost:9000"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.17.0
  - calendar: 4.1.0
  - cloud_federation_api: 1.8.0
  - comments: 1.15.0
  - contacts: 5.0.1
  - dav: 1.24.0
  - encryption: 2.13.0
  - federatedfilesharing: 1.15.0
  - files: 1.20.1
  - files_downloadactivity: 1.15.0
  - files_fulltextsearch: 24.0.1
  - files_lock: 24.0.1
  - files_pdfviewer: 2.6.0
  - files_rightclick: 1.4.0
  - files_sharing: 1.17.0
  - files_trashbin: 1.15.0
  - files_versions: 1.18.0
  - fileslibreofficeedit: 1.1.0
  - firstrunwizard: 2.14.0
  - fulltextsearch: 24.0.0
  - fulltextsearch_elasticsearch: 24.0.1
  - group_everyone: 0.1.11
  - groupfolders: 13.1.0
  - holiday_calendars: 0.2.1
  - logreader: 2.10.0
  - lookup_server_connector: 1.13.0
  - notes: 4.6.0
  - notifications: 2.13.1
  - oauth2: 1.13.0
  - onlyoffice: 7.6.8
  - password_policy: 1.15.0
  - previewgenerator: 5.1.1
  - privacy: 1.9.0
  - provisioning_api: 1.15.0
  - quota_warning: 1.15.0
  - serverinfo: 1.15.0
  - settings: 1.7.0
  - tasks: 0.14.5
  - text: 3.6.0
  - theming: 2.0.1
  - twofactor_admin: 0.2.1
  - twofactor_backupcodes: 1.14.0
  - twofactor_email: 2.7.1
  - twofactor_gateway: 0.20.0
  - twofactor_nextcloud_notification: 3.5.0
  - twofactor_totp: 7.0.0
  - twofactor_webauthn: 1.0.0
  - updatenotification: 1.15.0
  - viewer: 1.9.0
  - workflowengine: 2.7.0
Disabled:
  - admin_audit
  - bruteforcesettings: 2.5.0
  - circles: 24.0.0
  - contactsinteraction: 1.5.0
  - dashboard: 7.4.0
  - duplicatefinder: 0.0.15
  - extract: 1.3.5
  - federation: 1.14.0
  - files_external
  - files_trackdownloads: 1.11.0
  - group_default_quota: 0.1.4
  - maps: 0.2.1
  - nextcloud_announcements: 1.13.0
  - photos: 1.6.0
  - recommendations: 1.3.0
  - related_resources: 1.0.3
  - sharebymail: 1.14.0
  - support: 1.7.0
  - survey_client: 1.12.0
  - suspicious_login
  - systemtags: 1.14.0
  - user_ldap
  - user_status: 1.4.0
  - weather_status: 1.4.0
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by skjnldsv**
Fix https://github.com/nextcloud/server/issues/35558
Regression from https://github.com/nextcloud/server/pull/34100

When search for files, clicking the "load more" button, changes the URL and the legacy files list triggers a navigation change.
I fixed the url change and added a sanity check in the legacy url change detection to potentially mitigate other unseen issues.

### CODE REVIEW

File: core/src/views/UnifiedSearch.vue [hunk 1]
```diff
@@ -396,7 +396,7 @@ export default {
 			emit('nextcloud:unified-search.close')
 		},
 
-		resetForm() {
+		onNavigationChange() {
 			this.$el.querySelector('form[role="search"]').reset()
```
Comment (szaimen) [inline core/src/views/UnifiedSearch.vue:400]: This does not seem to work in my testing: navigating into a folder e.g. does not clear the search.

**Review by szaimen — COMMENTED** [review]
Tested and works however one small remark

File: core/src/views/UnifiedSearch.vue (hunk 1 above)
Comment (szaimen) [reply core/src/views/UnifiedSearch.vue:400]: After checking the logic a bit more, the `files:navigation:changed` does not seem to get triggere when navigating into a folder? So it seems to work as intended...

**Review by szaimen — APPROVED** [review]
see above

File: core/src/views/UnifiedSearch.vue (hunk 1 above)
Comment (szaimen) [reply core/src/views/UnifiedSearch.vue:400]: Would it be hard to implement a logic to also reset it when navigating into a folder?

Comment (skjnldsv) [reply core/src/views/UnifiedSearch.vue:400]: Hum, I'll think about it yeah

Comment (skjnldsv) [reply core/src/views/UnifiedSearch.vue:400]: Ah wait, this was never implemented anyway. This was about changing the view in the navigation, so the current behaviour is actually correct. I misunderstood what you meant :)

We can open a dedicated ticket to make the search reset on dir change sure! Not sure if that was intentional or not. Can't recall

Comment (skjnldsv) [reply core/src/views/UnifiedSearch.vue:400]: https://github.com/nextcloud/server/issues/38312

**Review by artonge — APPROVED** [review]

**Comment by ostasevych** [comment]
Hi! Please, suggest quick and dirty hack how to commit these changes on my personal prod site?

**Comment by skjnldsv** [comment]
> Hi! Please, suggest quick and dirty hack how to get these changes on my personal prod site?

You can download the patch and manually try to apply it from either the stable25 or stable26 branch
- https://github.com/nextcloud/server/pull/38325
- https://github.com/nextcloud/server/pull/38318

https://coderwall.com/p/6aw72a/creating-patch-from-github-pull-request

**Comment by t-markmann** [comment]
> > Hi! Please, suggest quick and dirty hack how to get these changes on my personal prod site?
> 
> You can download the patch and manually try to apply it from either the stable25 or stable26 branch
> 
>     * [[stable25] fix(search): fix load more #38325](https://github.com/nextcloud/server/pull/38325)
> 
>     * [[stable26] fix(search): fix load more #38318](https://github.com/nextcloud/server/pull/38318)
> 
> 
> https://coderwall.com/p/6aw72a/creating-patch-from-github-pull-request

Unfortunately this only seems to work, if your (production) NC instance is a git repo. 
If you execute the .patch in a normally installed NC directory, this will be returned: fatal: not a git repository (or any parent up to mount point /)

Solution might be "git init" and so on. But then your NC dir ist not clean and upgradeable anymore, I think.

A more practicable solution: download the 3 (non-src) files and replace the old ones. 
These commands will do (if you adjust your www-data user and the nc path):

```
NCBASE=/var/www/nextcloud

cd ${NCBASE}/apps/files/js
sudo -u www-data curl -O https://raw.githubusercontent.com/nextcloud/server/b4d7f56b7adb5f74b243f3520b774451158c57d5/apps/files/js/app.js

cd ${NCBASE}/dist
sudo -u www-data curl -O https://raw.githubusercontent.com/nextcloud/server/b4d7f56b7adb5f74b243f3520b774451158c57d5/dist/core-unified-search.js

sudo -u www-data curl -O https://raw.githubusercontent.com/nextcloud/server/b4d7f56b7adb5f74b243f3520b774451158c57d5/dist/core-unified-search.js.map
```

**Comment by ostasevych** [comment]
> > > Hi! Please, suggest quick and dirty hack how to get these changes on my personal prod site?
> > 
> > 
> > You can download the patch and manually try to apply it from either the stable25 or stable26 branch
> > ```
> > * [[stable25] fix(search): fix load more #38325](https://github.com/nextcloud/server/pull/38325)
> > 
> > * [[stable26] fix(search): fix load more #38318](https://github.com/nextcloud/server/pull/38318)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > https://coderwall.com/p/6aw72a/creating-patch-from-github-pull-request
> 
> Unfortunately this only seems to work, if your (production) NC instance is a git repo. If you execute the .patch in a normally installed NC directory, this will be returned: fatal: not a git repository (or any parent up to mount point /)
> 
> Solution might be "git init" and so on. But then your NC dir ist not clean and upgradeable anymore, I think.
> 
> A more practicable solution: download the 3 (non-src) files and replace the old ones. These commands will do (if you adjust your www-data user and the nc path):
> 
> ```
> NCBASE=/var/www/nextcloud
> 
> cd ${NCBASE}/apps/files/js
> sudo -u www-data curl -O https://raw.githubusercontent.com/nextcloud/server/b4d7f56b7adb5f74b243f3520b774451158c57d5/apps/files/js/app.js
> 
> cd ${NCBASE}/dist
> sudo -u www-data curl -O https://raw.githubusercontent.com/nextcloud/server/b4d7f56b7adb5f74b243f3520b774451158c57d5/dist/core-unified-search.js
> 
> sudo -u www-data curl -O https://raw.githubusercontent.com/nextcloud/server/b4d7f56b7adb5f74b243f3520b774451158c57d5/dist/core-unified-search.js.map
> ```

By replacing these 3 files I am obtaining empty home folder with this error in the logs:
`TypeError: OCA\Files\Service\TagService::__construct(): Argument #4 ($homeFolder) must be of type OCP\Files\Folder, null given, called in /var/www/html/nextcloud/apps/files/lib/AppInfo/Application.php on line 111` :(

**Comment by t-markmann** [comment]
Hm, I'm running a NC 25.0.5.1
Works fine here...

**Comment by ostasevych** [comment]
> Hm, I'm running a NC 25.0.5.1 Works fine here...

With 26.0.1 it goes not so well. I haven't to revert back.

**Comment by t-markmann** [comment]
The Files for 25 / 26 are different. 
Try these for NC26:

```
NCBASE=/var/www/nextcloud

cd ${NCBASE}/apps/files/js
sudo -u www-data curl -O https://raw.githubusercontent.com/nextcloud/server/6da7e66dc19696f991968bb36fba0ee3a3982078/apps/files/js/app.js

cd ${NCBASE}/dist
sudo -u www-data curl -O https://raw.githubusercontent.com/nextcloud/server/6da7e66dc19696f991968bb36fba0ee3a3982078/dist/core-unified-search.js

sudo -u www-data curl -O https://raw.githubusercontent.com/nextcloud/server/6da7e66dc19696f991968bb36fba0ee3a3982078/dist/core-unified-search.js.map
```

**Comment by ostasevych** [comment]
> The Files for 25 / 26 are different. Try these for NC26:
> 
> ```
> NCBASE=/var/www/nextcloud
> 
> cd ${NCBASE}/apps/files/js
> sudo -u www-data curl -O https://raw.githubusercontent.com/nextcloud/server/6da7e66dc19696f991968bb36fba0ee3a3982078/apps/files/js/app.js
> 
> cd ${NCBASE}/dist
> sudo -u www-data curl -O https://raw.githubusercontent.com/nextcloud/server/6da7e66dc19696f991968bb36fba0ee3a3982078/dist/core-unified-search.js
> 
> sudo -u www-data curl -O https://raw.githubusercontent.com/nextcloud/server/6da7e66dc19696f991968bb36fba0ee3a3982078/dist/core-unified-search.js.map
> ```

Thanks. That works fine!

**Comment by skjnldsv** [comment]
> Unfortunately this only seems to work, if your (production) NC instance is a git repo.

Yeah, this is not for production. 
If you don't know what you're doing, I suggest to not experiment with your production services :)

**Pull request merged**